src/users_db.rs

```diff
@@ -150,9 +163,45 @@ impl UsersDb {
         )
     }
 
-    pub fn read(&self) -> rusqlite::Result<Vec<User>> {
-        let mut stmt = try!(self.connection.prepare("SELECT * FROM users"));
-        let rows = try!(stmt.query(&[]));
+    pub fn read(&self, filter: ReadFilter) -> rusqlite::Result<Vec<User>> {
+        let mut stmt = try!(
+            self.connection.prepare("SELECT * FROM users")
```

Comment:
There's a funny compiler error above `stmt` not living long enough if I remove `stmt` from that scope. I could prepare an empty query instead.